ci/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"dagger.io/dagger"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// create a Dagger client
+	fmt.Println("Initializing Dagger...")
+	client, err := dagger.Connect(ctx)
+	if err != nil {
+		panic(err)
+	}
+	defer client.Close()
+
+	src := client.Host().Directory(".") // TODO: excludes
+
+	fmt.Println("Executing CI...")
+	builder, err := builder(client, src).Sync(ctx)
+	if err != nil {
+		fmt.Printf("Error creating CI environment: %+v\n", err)
+		panic(err)
+	}
+
+	builder = builder.WithExec([]string{"go", "mod", "tidy"}).
+		WithExec([]string{"go", "build"}).
+		WithExec([]string{"go", "test", "./..."})
```

Comment:
in the original pipeline, this would call `./script/test.sh`, which also appends to `$GITHUB_STEP_SUMMARY`... not sure if that's possible with dagger too..

Reply:
Yeah that should be possible, I'll try it out